http-server/auth/auth.go

```diff
@@ -0,0 +1,159 @@
+			return []byte(envPassword), nil
+		})
+
+		if err == nil && token.Valid && claims.ExpiresAt > time.Now().Unix() {
+			response := map[string]string{"token": tokenStr}
+			answear, err := json.Marshal(response)
+			if err != nil {
+				msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"sirelization error"}, false)
+				w.WriteHeader(errInt)
+				w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+				w.Write(msg)
+				return
+			}
+			w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+			w.Write(answear)
+			return
+		}
+	}
+	// генерация нового JWT-токена
+	expirationTime := time.Now().Add(8 * time.Hour)
+	claims := &Claims{
+		Authorized: true,
+		Sum:        500,
+		StandardClaims: jwt.StandardClaims{
+			ExpiresAt: expirationTime.Unix(),
+		},
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	tokenString, err := token.SignedString([]byte(envPassword))
+	if err != nil {
+		msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"error when creating a new token"}, false)
+		w.WriteHeader(errInt)
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write(msg)
+		return
+	}
+	http.SetCookie(w, &http.Cookie{
+		Name:     "token",
+		Value:    tokenString,
+		Path:     "/",
+		Expires:  expirationTime,
+		HttpOnly: true,
+	})
+	// записываем в формате json JWT-токен и отправляем пользователю.
+	response := map[string]string{"token": tokenString}
+	answear, err := json.Marshal(response)
+	if err != nil {
+		msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"sirelization error"}, false)
+		w.WriteHeader(errInt)
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.Write(msg)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	w.Write(answear)
+}
+
+// CheckToken function for user authentication.
+// CheckToken функция для аутентификации пользователей.
+func CheckToken(next http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		envPassword := os.Getenv("TODO_PASSWORD")
+		// парсим JWT-токен из куки и если он действителен пропускаем на следующую страничку
+		if len(envPassword) > 0 {
+			var cookieToken string
+			cookie, err := req.Cookie("token")
+			if err != nil {
+				msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"unauthorized user"}, true)
+				w.WriteHeader(errInt)
+				w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+				w.Write(msg)
+				return
+			}
+			cookieToken = cookie.Value
+			claims := &Claims{}
+			token, err := jwt.ParseWithClaims(cookieToken, claims, func(token *jwt.Token) (interface{}, error) {
+				if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+					return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+				}
+				return []byte(envPassword), nil
+			})
+			if err != nil {
```

Comment:
Кажется, что можно объединить
```suggestion
			if err != nil || !token.Valid {
```